src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -241,6 +242,12 @@ public StudyParticipant getParticipant(Study study, String userId, boolean inclu
         return getParticipant(study, account, includeHistory);
     }
     
+    public StudyParticipant getParticipant(Study study, String healthCode) {
+        AccountId accountId = AccountId.forHealthCode(study.getIdentifier(), healthCode);
+        Account account = getAccountThrowingException(accountId);
+        return getParticipant(study, account, false);
```

Comment:
I'm not sure why you need this since ParticipantController.getActivityEventsForWorker() calls ParticipantService.getActivityEvents() and that already loads an Account. Probably you want to remove the activityEventService.getActivityEventList() method that takes a healthCode, and just pass that Account to it... ?